src/node.rs

```diff
@@ -112,10 +119,27 @@ impl Node {
 
         Some((class_name, class_version))
     }
+
+    pub fn cmp_by_id(&self, other: &Node) -> cmp::Ordering {
+        self.get_id().cmp(&other.get_id())
+    }
+
+    pub fn get_basic_as_str(&self) -> &'static str {
+        match NodeBasic::from_u8(self.get_basic()) {
+            Some(NodeBasic::Controller) => "Controller",
+            Some(NodeBasic::StaticController) => "Static Controller",
+            Some(NodeBasic::Slave) => "Slave",
+            Some(NodeBasic::RoutingSlave) => "Routing Slave",
+            _ => "unknown"
+        }
+    }
 }
 
 impl fmt::Debug for Node {
     fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        // The Z-Wave network is a mesh, so don't print neighbors. You'll quite
+        // possibly wind up with infinite output.
```

Comment:
I think that we could print just the node ids of the neighbors as long as we don't use {:?} on the neigbors vec